packages/worker/src/api/controllers/global/users.ts

```diff
@@ -145,12 +145,19 @@ export const adminUser = async (
       )
     }
 
+    // Get the first and last names from the SSO display name if available
+    const nameParts = (displayName ?? "").split(" ").filter((p: string) => !!p)
+    const firstName = nameParts.length > 0 ? nameParts[0] : ""
```

Comment:
This might be conflicting. This might be the most common case, but not the only one. We will have names in multiple languages, and names such as `Mary Jane Doe`. Not sure if we should be inferring the name from the display name.
It could be that the display name is also a nickname or whatever, giving us even weirder results

Reply:
SSO can provide optional meta with the first and second name. Maybe we should check if these are populated depending on the used SSO

Reply:
This definitely seems like a better approach, I was copying some older code. There are some other places that use this code as well which I can update too.

Reply:
Updated to pass in the givenName and familyName which is provided by both Google and Microsoft SSO.